rotkehlchen/globaldb/handler.py

```diff
@@ -208,6 +209,140 @@ def add_asset(
+
+        return assets_list, entries_found
+
+    @staticmethod
+    def get_assets_mappings(identifiers: List[str]) -> List[Dict[str, str]]:
+        """Given a list of asset identifiers, return a list of asset information(id, name, symbol)
+        for those identifiers.
+        May raise:
+        - InputError if one of the identifiers do not exist
+        """
+        result = []
+        identifiers_query = f'assets.identifier IN ({",".join("?" * len(identifiers))})'
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT assets.identifier, assets.name, common_asset_details.symbol FROM assets '
+                'JOIN common_asset_details on assets.identifier = common_asset_details.identifier '
+                'WHERE ' + identifiers_query,
+                tuple(identifiers),
+            )
+            for entry in cursor:
+                result.append({
+                    'identifier': entry[0],
+                    'name': entry[1],
+                    'symbol': entry[2],
+                })
+            if len(result) != len(identifiers):
+                raise InputError('One or more identifiers do not exist')
+        return result
+
+    @staticmethod
+    def search_assets(filter_query: 'AssetsFilterQuery') -> List[Dict[str, str]]:
+        search_result = []
+        query, bindings = filter_query.prepare()
+        parent_query = """
+        SELECT assets.identifier, name, symbol FROM assets
+        JOIN common_asset_details on assets.identifier = common_asset_details.identifier
+        """
+        query = parent_query + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                search_result.append({
+                    'identifier': entry[0],
+                    'name': entry[1],
+                    'symbol': entry[2],
+                })
+        return search_result
+
+    @staticmethod
+    def search_assets_levenshtein(
+            filter_query: 'AssetsFilterQuery',
+            substring_search: str,
+            limit: Optional[int],
+    ) -> List[Dict[str, str]]:
+        search_result = []
+        query, bindings = filter_query.prepare()
+        parent_query = """
+        SELECT assets.identifier, name, symbol FROM assets
+        JOIN common_asset_details on assets.identifier = common_asset_details.identifier
+        """
+        query = parent_query + query
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            cursor.execute(query, bindings)
+            for entry in cursor:
+                lev_dist_name = levenshtein_distance(substring_search, entry[0])
+                lev_dist_symbol = levenshtein_distance(substring_search, entry[1])
+                lev_dist_average = (lev_dist_name + lev_dist_symbol) / 2
+                if lev_dist_average <= 4:
```

Comment:
Why is 4 here? Is the number of different chars?

Reply:
but how was 4 choosen?

Reply:
but how was 4 choosen?

Reply:
Magic number ;)
4 ->  on average, I'd have to make 4 manipulations(addition, deletion, editing) for string 1 to match string 2.

Reply:
so yes, having this a constant with an explanation somewhere is better than leaving it as it is